app/src/main/java/com/example/vybe/ViewProfileActivity.java

```diff
@@ -2,16 +2,58 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 
+import android.content.Intent;
 import android.os.Bundle;
+import android.widget.Button;
+import android.widget.TextView;
+
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
 
 /**
  * This activity still does nothing but will be fixed later
  */
 public class ViewProfileActivity extends AppCompatActivity {
 
+    private TextView usernameTextView;
+    private TextView emailTextView;
+    private Button logoutBtn;
+
+    private User user;
+
+    private FirebaseAuth mAuth = FirebaseAuth.getInstance();
+    private FirebaseFirestore db = FirebaseFirestore.getInstance();
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_view_profile);
+
+        usernameTextView = findViewById(R.id.username_profile);
+        emailTextView = findViewById(R.id.email_profile);
+        logoutBtn = findViewById(R.id.logout_btn);
+
+        Intent intent = getIntent();
+        Bundle extras = intent.getExtras();
+
+        if (extras.containsKey("user")) {
+            user = (User) extras.getSerializable("user");
+            usernameTextView.setText(user.getUsername());
+            emailTextView.setText(user.getEmail());
+        }
+//        db.collection("Users").document(mAuth.getCurrentUser().getUid())
+//                .get()
+//                .addOnSuccessListener((DocumentSnapshot doc) -> {
+//                    usernameTextView.setText((String) doc.getData().get("username"));
+//                    emailTextView.setText((String) doc.getData().get("email"));
+//                });
```

Comment:
Can this be removed?

Reply:
If something is to get done later, pls make a new issue for it @jacobwong98

Reply:
Removed